### ISSUE TITLE
[Feature]: Add method to remove sequence match info from cigar

### ISSUE THREAD

**Issue by MillironX** [BioJulia/BioAlignments.jl#90]
## Expected Behavior

I would like a way to tell if the two _alignments_ are the same regardless of sequence, such that

```julia
Alignment("1=1X") == Alignment("2M")
```

## Current Behavior

```julia
Alignment("1=1X") == Alignment("2M")

# returns false
```

:point_up: This behavior is correct!

## Possible Solution / Implementation

The current behavior is entirely correct, but it doesn't let me compare alignments that have matching against those that don't. I propose a new function

```julia
remove_match_ops(::T) where {T<:Union{String,Alignment,AlignedSequence,PairwiseAlignment,PairwiseAlignmentResult}}
```

that would remove the `=` (sequence match) and `X` (sequence mismatch) operations from the CIGAR of the alignment and return a `T` where those operations would be replaced by `M` (match) operations and adjacent match operations merged.

**Comment by kescobo** [comment]
I don't work much with CIGAR, but I can't think of an objection

**Comment by cvigilv** [comment]
Wouldn't it be better to implement a CIGARString type that can have the methods for this type of comparison? That way, one could expose a string macro `cigar` to generate the CIGARString struct based on a CIGAR string and then use the underlying methods for this, skipping the need to add this behaviour to the `Alignment` structure.

Example code of how it could look:
```julia
cigar"1=1X" == cigar"2M"
```

**Comment by kescobo** [comment]
Yeah, conveniently we have https://github.com/BioJulia/CIGARStrings.jl now - not sure whether this kind of equality is implemented there - @jakobnissen ?

**Comment by jakobnissen** [comment]
It isn’t, but I think it’s a good idea. I’ll add it to CigarStrings, and then it can be a separate issue to use CigarStrings.jl in this package

**Comment by jakobnissen** [comment]
@kescobo CIGARStrings.jl version 0.1.5 has now been released with a function that does this. It also handles `OP_S` versus `OP_H` (same semantic meaning), skips `OP_P` (no semantic meaning) and treats e.g. `2D` and `1D1D` as equivalent.

**Comment by jakobnissen** [comment]
CIGARStrings v0.1.7 is now being registered, which contains a function `normalize`, which does exactly as asked.
